doc/anatomy_of_an_extension.md

```diff
@@ -0,0 +1,236 @@
+        "value": "Apache 2.0"
+    },
+    {
+        "name": "Custom detail",
+        "value": "Value for custom detail"
+    }
+]
+```
+
+There is also a `components` object that should contain an array of all the components included in the extension. For example: 
+```json
+"components": [
+    "my_custom_component",
+    "another_custom_component"
+]
+```
+___
+
+### Components
+
+Components should aim to cover specific functionality, like adding a new column with a UUID, running an analysis and storing the result on a separate row or sending an HTTP request to a specific endpoint. 
+
+Tipically, a component receives one or more inputs; it has some settings that influence the execution of the code; and produce an output. 
+
+In Workflows, most components produce a table that contains the same columns from the input plus an additional set of columns that contain the result. This is not a hard requirement though, and your component doesn't need to follow this pattern.
+
+Each component should be created on a separate folder inside [`/components`](../components/) and it's defined by **metadata** and **logic** (implemented as a stored procedure). 
+#### Component's metadata
+Each component has its own [`metadata.json`](../components/template/metadata.json) file, where you can define a name, category, description, icon, etc. And most importantly, **inputs**, **outputs** and a link with the corresponding stored procedure.
+
+Find more information about the component's metadata in the specific [documentation](./component_metadata.md).
+
+#### Logic
+The logic for each component is defined as an stored procedure in the [`components/<component_name>/src/procedure.sql`](../components/template/src/procedure.sql) file.
+
+Find a more complete documentation about creating stored procedures for custom components in [this documentation](./procedure.md).
+
+#### Joining metadata and logic
+Inputs, settings, and outputs of a component are defined in the metadata file, and then used in the stored procedure. 
+
+For these two parts of a custom component to work well together, we just need to ensure consistency between what's declared in metadata and what's used in the stored procedure.
+
+For this example we'll create a very simple component that just adds a new column with a fixed value.
+
+##### Inputs
+As mentioned before, inputs defined in the metadata need to match the inputs of the stored procedure. For example: 
+```json
+"inputs": [
+    {
+      "name": "input_table",
+      "title": "Input A",
+      "description": "A table to be used as input for my component",
+      "type": "Table"
+    },
+    {
+      "name": "value",
+      "title": "Value",
+      "description": "A value that will be used to populate the new column",
+      "type": "String"
+  }
+]
+```
+
+the procedure should be declared as follows: 
+```sql
+CREATE OR REPLACE PROCEDURE ADD_FIXED_VALUE_COLUMN(
+  input_table STRING, 
+  value STRING
+)
+BEGIN
+    EXECUTE IMMEDIATE '''
```

Comment:
I think this is confusing. A component without an output will be basically useless. It can run a SELECT query (like in your example), but that will do nothing, the response from that query will be lost and not used anywhere. Maybe better to introduce inputs and outputs together, and then add the simple example